### PR TITLE
Do not insert inline images inside code blocks

### LINKED ISSUE #9567: It's possible to create a code block with an image inside
## Steps to reproduce:
1. Add some text with inline image inside
2. Select everything
3. Click `Insert code block`

## Result:
A code block with image inside is created

## Expected:
Code block cannot contain images

## Screencast:

https://user-images.githubusercontent.com/34380544/115726501-113a2a80-a383-11eb-9e85-c56ec4ca5354.mp4

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (code-block): The code block feature will not allow for inserting inline images as its content. Closes #9567.

### CODE REVIEW

**Review by oleq — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-code-block/src/codeblockediting.js [hunk 1]
```diff
@@ -228,6 +228,15 @@ export default class CodeBlockEditing extends Plugin {
 			data.preventDefault();
 			evt.stop();
 		}, { context: 'pre' } );
+
+		// Disallow `imageInline` inside `codeBlock`. See #9567.
+		if ( editor.plugins.has( 'ImageInlineEditing' ) ) {
+			editor.model.schema.addChildCheck( ( context, childDefinition ) => {
+				if ( context.endsWith( 'codeBlock' ) && childDefinition.name == 'imageInline' ) {
```
[the remark this answers is not in the document]
Comment (pomek) [reply packages/ckeditor5-code-block/src/codeblockediting.js:235]: Yeah, I wasn't sure about it, but I couldn't figure out another use case. I'll improve it.

[the remark this answers is not in the document]
Comment (pomek) [reply packages/ckeditor5-code-block/src/codeblockediting.js]: @oleq, fixed.

File: packages/ckeditor5-code-block/src/codeblockcommand.js [hunk 2]
```diff
@@ -115,6 +115,11 @@ export default class CodeBlockCommand extends Command {
 			writer.rename( block, 'codeBlock' );
 			writer.setAttribute( 'language', language, block );
 			schema.removeDisallowedAttributes( [ block ], writer );
+
+			// The `codeBlock` should contain only $text nodes as its children. See #9567.
```
Comment (pomek) [inline packages/ckeditor5-code-block/src/codeblockcommand.js:119]:
```suggestion
			// The `codeBlock` element should contain only $text nodes as its children. See #9567.
```

File: packages/ckeditor5-code-block/src/codeblockcommand.js [hunk 3]
```diff
@@ -115,6 +115,11 @@ export default class CodeBlockCommand extends Command {
 			writer.rename( block, 'codeBlock' );
 			writer.setAttribute( 'language', language, block );
 			schema.removeDisallowedAttributes( [ block ], writer );
+
+			// The `codeBlock` element should contain only $text nodes as its children. See #9567.
+			Array.from( block.getChildren() )
+				.filter( child => !child.is( '$text' ) )
+				.forEach( writer.remove.bind( writer ) );
```
Comment (niegowski) [inline packages/ckeditor5-code-block/src/codeblockcommand.js:122]: This is removing also the soft breaks: 



We should use schema to check for allowed children:

```suggestion
			// The `codeBlock` element should contain only $text nodes as its children. See #9567.
			Array.from( block.getChildren() )
				.filter( child => !schema.checkChild( block, child ) )
				.forEach( child => writer.remove( child ) );
```

**Review by niegowski — CHANGES_REQUESTED** [review]

**Review by niegowski — APPROVED** [review]

**Pull request merged**